### PR TITLE
Fix reuseApplication when ws URI does not end with `/`

### PR DESCRIPTION
**Pull request by dnfield**
Another one on my journey to reland the GC benchmark stuff. This was one of the causes of failures on CI for #82005

### CODE REVIEW

File: packages/flutter_tools/lib/src/drive/drive_service.dart [hunk 1]
```diff
@@ -198,7 +198,7 @@ class FlutterDriverService extends DriverService {
   ) async {
     Uri uri;
     if (vmServiceUri.scheme == 'ws') {
-      uri = vmServiceUri.replace(scheme: 'http', path: vmServiceUri.path.replaceFirst('ws/', ''));
+      uri = vmServiceUri.replace(scheme: 'http', path: vmServiceUri.path.replaceFirst('/ws', ''));
```
Comment (dnfield) [inline packages/flutter_tools/lib/src/drive/drive_service.dart:201]: I don't fully understand why we even do this - does DDS expect an HTTP uri?

Comment (dnfield) [reply packages/flutter_tools/lib/src/drive/drive_service.dart:201]: (in the end we convert this back to a WebSocket URI anyway to connect)

Comment (dnfield) [reply packages/flutter_tools/lib/src/drive/drive_service.dart:201]: Fixed up this patch for that case and added a test.

**Review by jonahwilliams — APPROVED** [review]
LGTM

File: packages/flutter_tools/lib/src/drive/drive_service.dart [hunk 2]
```diff
@@ -198,7 +198,9 @@ class FlutterDriverService extends DriverService {
   ) async {
     Uri uri;
     if (vmServiceUri.scheme == 'ws') {
-      uri = vmServiceUri.replace(scheme: 'http', path: vmServiceUri.path.replaceFirst('ws/', ''));
```
[the remark this answers is not in the document]
Comment (dnfield) [reply packages/flutter_tools/lib/src/drive/drive_service.dart:201]: (talked offline - that line is going from HTTP back to WS)

**Pull request merged**